### PR TITLE
Issue 3381 Android app bundle fixes

### PR DESCRIPTION
**Pull request by britzl**
* Fixed issues when bundling projects without native extensions
* Some code cleanup

### CODE REVIEW

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidAAB.java [hunk 1]
```diff
@@ -172,37 +172,44 @@ private static void copyEngineBinary(Project project, Platform architecture, Fil
...
+			else {
+				for (File dir : packagesDir.listFiles(File::isDirectory)) {
+					File resDir = new File(dir, "res");
+					if (resDir.isDirectory()) {
+						directories.add(resDir);
+					}
 				}
 			}
 		}
+		else {
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidAAB.java:203]: Copy resources (icons) in non-ne builds!

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/BundleHelper.java [hunk 2]
```diff
@@ -451,22 +451,20 @@ private File getAndroidResourceDir(File appDir) {
...
     }
 
-    public void copyAndroidResources(Platform platform, File appDir) throws IOException {
-        boolean hasExtensions = ExtenderUtil.hasNativeExtensions(project);
+    public File copyAndroidResources(Platform platform) throws IOException {
+        if (ExtenderUtil.hasNativeExtensions(project)) {
+            return null;
+        }
 
-        File targetResDir = getAndroidResourceDir(appDir);
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/BundleHelper.java:457]: Unused!

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/BundleHelper.java [hunk 3]
```diff
@@ -451,22 +451,20 @@ private File getAndroidResourceDir(File appDir) {
         return new File(appDir, "res");
     }
 
-    public void copyAndroidResources(Platform platform, File appDir) throws IOException {
-        boolean hasExtensions = ExtenderUtil.hasNativeExtensions(project);
+    public File copyAndroidResources(Platform platform) throws IOException {
+        if (ExtenderUtil.hasNativeExtensions(project)) {
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/BundleHelper.java:455]: Exit early if NE build

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/BundleHelper.java [hunk 4]
```diff
@@ -451,22 +451,20 @@ private File getAndroidResourceDir(File appDir) {
...
-            resDir.mkdirs();
+        File resDir = new File(packagesDir, "com.defold.android/res");
+        resDir.mkdirs();
 
-            BundleHelper.createAndroidResourceFolders(resDir);
-            copyAndroidIcons(resDir);
-        }
+        BundleHelper.createAndroidResourceFolders(resDir);
+        copyAndroidIcons(resDir);
+        return resDir;
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/BundleHelper.java:467]: Return dir where resources were copied

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidAAB.java [hunk 5]
```diff
@@ -172,37 +172,44 @@ private static void copyEngineBinary(Project project, Platform architecture, Fil
...
-		if (packagesList.exists()) {
-			List<String> allLines = Files.readAllLines(new File(packagesDir, "packages.txt").toPath());
-			for (String line : allLines) {
-				File resDir = new File(packagesDir, line);
-				if (resDir.isDirectory()) {
-					directories.add(resDir);
+		if(packagesDir.exists()) {
+			File packagesList = new File(packagesDir, "packages.txt");
+			if (packagesList.exists()) {
+				List<String> allLines = Files.readAllLines(new File(packagesDir, "packages.txt").toPath());
```
Comment (JCash) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidAAB.java:186]: -> `Files.readAllLines(packagesList.toPath());`

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidAAB.java (hunk 1 above)
Comment (JCash) [reply com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidAAB.java:203]: Perhaps would be good to add that comment in the code?

**Review by JCash — DISMISSED** [review]
Looks good!

**Pull request merged**